### PR TITLE
(chore) upgrade to rc5

### PR DESCRIPTION
**Pull request by nadavsinai**
- update dependencies
 - add appModule,
 - fix bootstrap
 - remove boilerplate from inner components

### CODE REVIEW

**Comment by nadavsinai** [comment]
I signed it!

**Comment by nadavsinai** [comment]
I offer to make another PR where github route will be pulled to its own module, with router async loading of this module and child routes, however since this is webpack based and router has dependency on SystemJS loader it will require upgrading to webpack 2 (2.1.beta.20, and adding a runtime of System polyfill (System.js), anyone intrested in this?

File: package.json [hunk 1]
```diff
@@ -22,19 +22,23 @@
     "raw-loader": "^0.5.1",
     "to-string-loader": "^1.1.4",
     "typescript": "~1.8.9",
-    "typings": "~1.0.3",
+    "typings": "~1.3.2",
```
Comment (pkozlowski-opensource) [inline package.json]: Is it a mandatory change to update to RC5? If not please revert.

File: package.json [hunk 2]
```diff
@@ -22,19 +22,23 @@
...
     "webpack": "^1.12.9",
     "webpack-dev-server": "^1.14.0",
-    "webpack-merge": "^0.8.4"
+    "webpack-merge": "^0.14.1"
```
Comment (pkozlowski-opensource) [inline package.json]: Is it a mandatory change to update to RC5? If not please revert.

File: package.json [hunk 3]
```diff
@@ -22,19 +22,23 @@
...
-    "@angular/router": "3.0.0-beta.2",
+    "@angular/common": "2.0.0-rc.5",
+    "@angular/compiler": "2.0.0-rc.5",
+    "@angular/core": "2.0.0-rc.5",
+    "@angular/http": "2.0.0-rc.5",
+    "@angular/compiler-cli": "0.5.0",
+    "@angular/platform-server": "2.0.0-rc.5",
+    "@angular/forms": "0.3.0",
+    "@angular/platform-browser": "2.0.0-rc.5",
+    "typescript": "^1.9.0-dev",
```
Comment (pkozlowski-opensource) [inline package.json]: Is it a mandatory change to update to RC5? If not please revert.

File: src/app/about/about.ts [hunk 4]
```diff
@@ -1,10 +1,7 @@
-import {Component} from '@angular/core';
+import {Component} from "@angular/core";
```
Comment (pkozlowski-opensource) [inline src/app/about/about.ts]: Why this change? Please revert.

File: src/app/github/repo-browser/repo-browser.ts [hunk 5]
```diff
@@ -1,18 +1,16 @@
-import { Component } from '@angular/core';
-import { Router, ROUTER_DIRECTIVES } from '@angular/router';
-import { Github } from '../shared/github';
+import {Component} from "@angular/core";
+import {Router} from "@angular/router";
+import {Github} from "../shared/github";
```
Comment (pkozlowski-opensource) [inline src/app/github/repo-browser/repo-browser.ts]: Please revert quote changes.

**Comment by pkozlowski-opensource** [comment]
@nadavsinai thnx for the PR! I would gladly merge it but there are number of un-related changes in there and it is hard to see what is strictly needed for RC5. Could you please revert all the non-essential parts? Also, please don't change naming / formatting in this PR - we can discuss those in a separate issue / PR.

Would you be willing to make the above changes so I can merge this one?

File: package.json (hunk 1 above)
Comment (nadavsinai) [inline package.json]: not mandatory, just a updated version for typings, I thought it'll be a best practice to keep things up to date, will revert

File: package.json (hunk 2 above)
Comment (nadavsinai) [inline package.json]: same as typings above, mostly bugfixes in this, but can be a spearate PR if you'd like it.. will revert

File: package.json (hunk 3 above)
Comment (nadavsinai) [inline package.json]: here I need your guidance, the things that are new here apart from the rc5 update are : 
-   the inclusion of the forms module ( consider that when this seed was initially created the same functionality was included in common... I'd argue it needs to be in any seed
-   the inclusion of the compiler-cli module - to allow AoT compilation and it's dependent on typscript which is also included
 - the inclusion of platform-server for use of AoT compiler via node

so please guide me whether to omit some or all of these, the rc5 ver change obviously needs to to be kept..,

Comment (pkozlowski-opensource) [inline package.json]: Keep forms if we are using them. Let's not go into AoT in this PR. Let's focus on a minimal set of changes to get RC5 compatibility. Baby steps :-)

File: src/app/github/repo-browser/repo-browser.ts (hunk 5 above)
Comment (nadavsinai) [inline src/app/github/repo-browser/repo-browser.ts]: reverted

File: src/app/about/about.ts (hunk 4 above)
Comment (nadavsinai) [inline src/app/about/about.ts]: just IDE formatting... reverted, I removed unused imports and it did this...

File: package.json (hunk 3 above)
Comment (nadavsinai) [inline package.json]: hmm.. Ok another commit will be done.. will you want me to force push a rebased version to have only one commit ?

Comment (nadavsinai) [inline package.json]: done

**Comment by pkozlowski-opensource** [comment]
This looks good now! I will squash while merging. Thnx!
